pkg/bitcoin/chain.go

```diff
@@ -0,0 +1,31 @@
+package bitcoin
+
+// Chain defines an interface meant to be used for interaction with the
+// Bitcoin chain.
+type Chain interface {
+	// GetTransaction gets the transaction with the given transaction hash.
+	// If the transaction with the given hash was not found on the chain,
+	// this function returns an error.
+	GetTransaction(transactionHash Hash) (*Transaction, error)
+
+	// GetTransactionConfirmations gets the number of confirmations for the
+	// transaction with the given transaction hash. If the transaction with the
+	// given hash was not found on the chain, this function returns an error.
+	GetTransactionConfirmations(transactionHash Hash) (uint, error)
+
+	// BroadcastTransaction broadcasts the given transaction over the
+	// network of the Bitcoin chain nodes. If the broadcast action could not be
+	// done, this function returns an error. This function does not give any
+	// guarantees regarding transaction mining. The transaction may be mined or
+	// rejected eventually.
+	BroadcastTransaction(transaction *Transaction) error
+
+	// GetCurrentBlockNumber gets the number of the current block. If the
+	// current block was not determined, this function returns an error.
+	GetCurrentBlockNumber() (uint, error)
```

Comment:
@lukasz-zimnoch should it return the latest mined (confirmed) block or the block that is currently being mined?

Reply:
To be more precise maybe we could name this function `GetLatestBlockHeight`?